dev/src/v1/firestore_client.ts

```diff
@@ -347,9 +351,13 @@ export class FirestoreClient {
             }
             const func = stub[methodName];
             return func.apply(stub, args);
-          },
-        (err: Error | null | undefined) => () => {
-          throw err;
+          };
+        },
+        (err: Error | null | undefined) => {
+          this._stubFailed = true;
```

Comment:
Is there a way to make sure that setting "this._stubFailed = true" only happens once?

If for some reason the stub lives longer than expected and continues to experience errors, then the fresh reset stub will be reset by an old stub.

Also, this seems like an unexpected situation, and compensating behavior. Should we maybe add some logging to gain insight when this happens?

Reply:
The conditional block on ln 293, `if (this.firestoreStub ...) return this.firestoreStub` ensures that only one logical task will attempt to create a `firestoreStub`. The `firestoreStub` instance is a Promise, which will only resolve or reject once. Therefore the error handler on ln 357 will be called at max 1 time per `firestoreStub` instance. And we won't have two concurrent attempts to create the stub. So, I think this is safe.

Good idea on the logging. I will add some.

Reply:
logging added